conans/test/functional/generators/cmake_find_package_multi_test.py

```diff
@@ -261,3 +262,47 @@ def build(self):
                       "$<$<CONFIG:MinSizeRel>:;>;"
                       "$<$<CONFIG:Debug>:;>",
                       client.out)
+
+    @parameterized.expand([
+        ("find_package(hello 1.0)", True),
+        ("find_package(hello 1.1)", True),
+        ("find_package(hello 1.2)", False),
+        ("find_package(hello 1.0 EXACT)", False),
+        ("find_package(hello 1.1 EXACT)", True),
+        ("find_package(hello 1.2 EXACT)", False),
+        ("find_package(hello 0.1)", False),
+        ("find_package(hello 2.0)", False)
+    ])
+    def version_test(self, find_package_string, expected_result):
+        client = TestClient()
+        client.run("new hello/1.1 -s")
+        client.run("create .")
+
+        cmakelists = textwrap.dedent("""
+            cmake_minimum_required(VERSION 3.1)
+            project(consumer)
+            {find_package_string}
+            message(STATUS "hello found: ${{hello_FOUND}}")
+            """).format(find_package_string=find_package_string)
+
+        conanfile = textwrap.dedent("""
+            from conans import ConanFile, CMake
+
+
+            class Conan(ConanFile):
+                settings = "build_type"
+                requires = "hello/1.1"
+                generators = "cmake_find_package_multi"
+
+                def build(self):
+                    cmake = CMake(self)
+                    cmake.configure()
+            """)
+
+        client.save({"conanfile.py": conanfile, "CMakeLists.txt": cmakelists})
+        client.run("install .")
+        client.run("build .")
+        if expected_result:
+            self.assertIn("hello found: 1", client.out)
+        else:
+            self.assertIn("hello found: 0", client.out)
```

Comment:
shouldn't we expect a hard failure error? Not just continuing and printing "0", but having cmake really fail.

Reply:
with **CMake** and [find_package](https://cmake.org/cmake/help/v3.16/command/find_package.html) you always have a choice between hard and soft failures. e.g. the following:
```
find_package(Boost 1.70 REQUIRED)
```
will trigger a hard error (**cmake** command will fail and project won't be configured)
while the following:
```
find_package(Boost 1.70)
```
will only set **<Package>_FOUND** variable to the **FALSE**, allowing some user's fallback (e.g. choice between multiple libraries providing similar features, or just detect optional features).

Reply:
Good, thanks, makes sense.

I think this case of hard failure should also be covered by tests. Our problem is that REQUIRED is now implicit in our ``cmake_find_package`` generators, because if there is dependency the files are there, and if not, there are no files. A ``find_package()`` without REQUIRED, what does it mean? And if they are REQUIRED by default, shouldn't then the build fail, not warn?

Reply:
I am not sure I understand implicit `REQUIRED` -it's up to consumer to specify `REQUIRED` or omit it. in CMake it's only explicit - if you didn't specify it - it's not required. so it's totally optional and expected that package might be missing.
out of the conan, some CMake projects/libraries might be built both with and without conan, and might or might not put `REQUIRED`. even with conan, you're free to use several conanfiles or have optional dependencies and control options with profiles. like, some dependencies might be expected to be found on certain platform(s), but not others (e.g. Windows-only libraries).
in general, `find_package` without `REQUIRED` just means it's not fatal if package isn't here, like it provides some optional feature, or dependency might be found by some other means (embedded sources, downloaded from the CMake, etc.).

Reply:
Ok, yes, you are right.

Could you please add some tests with REQUIRED, to check they raise errors in the following cases:
- The package Config.cmake is not there
- The version doesn't match

Thanks!

Reply:
added more tests as requested